test/models/pipeline_run_test.rb

```diff
@@ -10,6 +10,10 @@ class PipelineRunTest < ActiveSupport::TestCase
   end
 
   test "duration is correct" do
-    assert_equal 0.0, pipeline_runs(:four).duration_hrs
+    assert pipeline_runs(:four).duration_hrs >= 0.0
+  end
+
+  test "precaching executes" do
+    pipeline_runs(:four).precache_report_info!
```

Comment:
What is the assertion for this test?
Also, based on the method implementation, this method doesn't use any info related to the pipeline_run model, since it pre-caches all reports. Looks like it should be a class method instead or be changed to cache a specific pipeline_run.

Reply:
> this method doesn't use any info related to the pipeline_run model

It uses the `pipeline_version` and the `sample` and others. See `report_info_params` method.

Reply:
I miss an `assert` line in this test. If someone puts a `return true` in that method this test will pass. If we don't have an assertion, this test can generate false positives, and should be removed from the test suite.

Reply:
Discussed offline. Will mock out.